Release 18.6.0

fheinecke opened the pull request:
### Identifier-first login enhancements
Teleport now automatically passes the username to the identifier provider when performing Identifier-first login with OIDC or SAML IdPs.

### GitHub Actions Kubernetes Wizard
Teleport now ships with a new guided flow for setting up GitHub Actions workflows that connects to Teleport-protected Kubernetes clusters without secrets.

### Other changes and improvements

* Fixed unspecified proxy address breaking moderated SFTP when mixing IPv4 and IPv6. [#62296](https://github.com/gravitational/teleport/pull/62296)
* Added full configuration file for `teleport-plugin-event-handler` helm chart. [#62280](https://github.com/gravitational/teleport/pull/62280)
* Added full environment variable configuration for event handler CLI. [#62280](https://github.com/gravitational/teleport/pull/62280)
* Added support for extraArgs/extraEnv/extraLabels patterns for `teleport-plugin-event-handler` helm chart. [#62266](https://github.com/gravitational/teleport/pull/62266)
* Fixed issue where AltGr key combinations did not work correctly in remote desktop sessions. [#62198](https://github.com/gravitational/teleport/pull/62198)
* Added `annotations` support for `teleport-plugin-event-handler` helm chart. [#62188](https://github.com/gravitational/teleport/pull/62188)
* Added a new global configuration section auth_connection_config allowing users to configure the backoff behavior for Proxy and Agent instances connecting to the Auth Service. [#62139](https://github.com/gravitational/teleport/pull/62139)
* Fixed a potential SSRF vulnerability in the Azure join method implementation. [#62406](https://github.com/gravitational/teleport/pull/62406)
* Support for v8 roles has been added to the Terraform provider. [#62380](https://github.com/gravitational/teleport/pull/62380)
* Added support for selecting Kube agents as Managed Updates v2 canaries. Important: the default update group is corrected to "default" from "stable/cloud". [#62211](https://github.com/gravitational/teleport/pull/62211)

Commit:
Release 18.6.0

Commit:
changelog fix

camscale commented:
As a new minor release, this should have the changes from https://goteleport.com/docs/upcoming-releases/ (with the tense changed) copied into the changelog.

aadc-dev committed:
updating CHANGELOG to reflect minor release format

aadc-dev committed:
pulling latest branch/v18 to pick up GHA wizard backport for v18.6 release

Merge branch 'branch/v18' of github.com:gravitational/teleport into release/18.6.0

aadc-dev committed:
pulling latest from v18, updating teleport.e reference

aadc-dev committed:
updating changelog
